packages/sign_in_with_apple/lib/src/authorization_request.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:meta/meta.dart';
+
+/// A base class which describes an authorization request that we will make on the native side.
+///
+/// Apple Docs: https://developer.apple.com/documentation/authenticationservices/asauthorizationrequest
+///
+/// We currently only support the following authorization requests:
+/// - [AppleIDAuthorizationRequest]
+/// - [PasswordAuthorizationRequest]
+@immutable
+abstract class AuthorizationRequest {
```

Comment:
@tp Please have a look here, especially on the documentation side

Reply:
Gladly, will do it first thing in the morning!